src/gesture-handler/fabricUtils.ts

```diff
@@ -0,0 +1,35 @@
+interface HostInstance {
+  _internalInstanceHandle: {
+    stateNode: {
+      node: Record<string, unknown>;
+    };
+  };
+  _nativeTag: number;
+}
+
+type localGlobal = typeof global & Record<string, unknown>;
+
+export function isFabric() {
+  return !!(global as localGlobal)._IS_FABRIC;
+}
+
+let findHostInstance: (ref: React.Component) => HostInstance | null = () => {
+  return null;
+};
+if (isFabric()) {
+  try {
+    findHostInstance =
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      require('react-native/Libraries/Renderer/shims/ReactFabric').findHostInstance_DEPRECATED;
+  } catch (e) {
+    throw new Error('[RNScreens] Cannot import `findHostInstance_DEPRECATED`.');
+  }
+}
```

Comment:
Will this work once we migrate completely to ESM? How will this conditional `require` work then?

Reply:
Should work without any problems, we have same code in Reanimated (https://github.com/software-mansion/react-native-reanimated/blob/main/src/reanimated2/fabricUtils.ts) and it works.

Reply:
AFAIK this won't emit proper ESM. @tboba you did some research on this topic with #1982, please take a look

Reply:
@kkafar Yup, as I checked previously how `lib` directory is being transpiled, this `require` declaration is being left as it is in ESM source code. As this could bring errors in some bundlers about mixed ESM/CJS code, I would consider if we should import specific packages like that.

Reply:
So, I understand that if we merge this in current form, #1982 is basically useless and ESM/CJS still will be mixed leading to errors with some bundlers, right?

Reply:
Most likely, but let me check that. I'm planning to test PR #1982 today with Vite bundler to check if this issue still appears - I'll also test this PR with this `require`.

Reply:
Thanks!

Reply:
@kkafar Unfortunately as I quickly checked how `lib` is being generated, all of the `require` statements are unfortunately not transpiled into `import` ones. I think we should find another way how we can import such optional dependencies (maybe with `await import`?) since some bundlers (like Vite) will still complain about mixed CJS/ESM code. 
However, I still can't manage to test that behavior - I've made a small example with @react-navigation/native-stack, Vite and react-native-web and I'm not getting any errors about the code 🤔 I think that I'm doing possibly something wrong, so I asked about the repro in the issue.

Reply:
I guess this file: `react-native/Libraries/Renderer/shims/ReactFabric` is always there, however I can see loading Fabric renderer on Paper leading to some issues. We need to check that & if there are no issues I'm fully in favour of importing it standard way.

Reply:
@WoLewicki the biggest problem is that `ReactFabric` only exists... on Fabric 😄 
I think that inline import would be possible here if this file could also exist on Paper - `if(isFabric())` is not without a reason there. But, as import has to be resolved from the global scope (it can't be imported after if or in any class/function, unless we are talking about the `await import`) it's not that easy though 😕